src/ansys_sphinx_theme/search/fuse_search.py

```diff
@@ -204,7 +204,26 @@ def create_search_index(app, exception):
 
     search_index_list = []
 
-    for document in app.env.found_docs:
+    static_search_options = app.config.html_theme_options.get("static_search", {})
+    excluded_docs = static_search_options.get("files_to_exclude", [])
+    included_docs = app.env.found_docs
+    output_dir = Path(app.builder.outdir)
+
+    for exclude_doc in excluded_docs:
+        exclude_doc = Path(exclude_doc)
+        exclude_path = output_dir / exclude_doc
+        is_folder = exclude_path.is_dir()
+
+        # Exclude documents based on whether exclude_doc is a folder or a file:
+        # - For folders, exclude all documents within the folder.
+        # - For files, exclude only the exact file match.
+        included_docs = [
+            doc
+            for doc in included_docs
+            if not (is_folder and Path(doc).parent == exclude_doc) and Path(doc) != exclude_doc
```

Comment:
@Revathyvenugopal162 Sorry but I'm not sure about the following LOC.
Let's say that you exlcude `my_folder`, doesn't that mean that `my_folder/subfolder/file` won't be ignored since it isn't the direct parent ?
Could we use `parents` instead to retrieve avoid that problem ?

Also, would it make sense to compute included_docs as follow:
```python
excluded_docs = static_search_options.get("files_to_exclude", [])
excluded_docs = [Path(document).resolve() for document in excluded_docs]
filtered_docs = []
for document in app.env.found_docs:
    document_path = Path(document).resolve()
    if document_path in excluded_docs:
        continue
    if any(parent.resolve() in excluded_docs for parent in document_path.parents):
        continue
    filtered_docs.append(document)
```

Reply:
@SMoraisAnsys For the moment, i implemented exlcude ``my_folder``, will exclude ``my_folder/subfolder/file`` as well. You wanted the other way ?